### PR TITLE
Create JavaScript article

### LINKED ISSUE #3: Write JavaScript article

### PR DESCRIPTION
**Pull request by ggliv**
Closes #3.

You can preview the rendered contents of this article [here](https://github.com/ggliv/DevDogs-Academy/blob/feature/js-article/frontend/javascript/README.md).

### CODE REVIEW

**Comment by justinquinnb** [comment]
Already said on Discord but commenting here too just for documentation's sake:

Fantastic work so far and excited to see where this article goes. Keep up the good work and give me or Avery a ping here/on Discord as you continue developing this. We really appreciate the hard work!

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:122]: Do we want to cover operations on `null` and `undefined` values?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:126]: I opted to omit `var` and no-keyword declarations since i see them as largely legacy features. Should they be included?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:153]: Should we include anything on variable shadowing?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:163]: I opted to omit mentions of the increment (`++`) and decrement (`--`) operators since I personally consider them to be a code smell. They'd also be a bit tough to describe in a concise way. Do we want to cover them?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:165]: Should we cover null coalescing and/or optional chaining?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:195]: I didn't talk about the string ordering comparisons since they're not terribly common and I didn't want to type out an explanation of lex order. Should they be touched on?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:328]: Are there any other show-stoppingly important array methods that should be included here?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:217]: I omitted switch statements since they're not very common. Should they be included?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:374]: Should we cover destructuring with for-of?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:377]: Should we talk about default parameters, parameter destructuring, varargs, generators, and/or closures?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:439]: Should we talk about getters and setters? (the JavaScript syntax sugar, not necessarily the broader concept of getters and setters)

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:475]: Should we include stuff on anonymous classes? Prototypes?

File: frontend/javascript/README.md
Comment (ggliv) [inline frontend/javascript/README.md:549]: We'll need to come back and make sure that these links are correct once their respective articles get written. (cc #1, #2)

**Review by ggliv — COMMENTED** [review]
I'm happy enough with the content covered in the current revision of the article, but here are some things I want to make sure are okay to omit (along with some general comments).

**Comment by ggliv** [comment]
The article still has some rough edges that I'll come back to sand down eventually, but the content is now there and largely in the shape I want it to be in.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:126]: As legacy features, I agree that there's little reason to include them. Let's see what @nivedhanatarajan7 thinks.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:163]: This I will leave entirely for @nivedhanatarajan7 to decide on. As someone mostly experienced in backend languages like Java and C++, increment/decrement operators seem like an important inclusion, but it sounds like they may work differently in JS.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:122]: Unless they produce a behavior different than what someone with basic CS knowledge might expect (e.g. a value instead of an error), I think this omission is alright.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:377]: I would argue for varargs and default params, but I can't speak to any of those other functionalities. Perhaps Avery or Nivedha have a stance on them.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:153]: Can't speak to this, will leave for Avery or Nivedha.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:439]: If JS has a shorthand for them or a very different way of writing them than Java/other introductory languages, I would argue for including getters and setters in JS.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md]: Noted.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:475]: Will leave for Nivedha or Avery's input.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:374]: Can't speak to this, will leave for Nivedha and/or Avery.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:217]: Since they're not commonly used in the language, I agree that their omission is justified.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:165]: Can't speak to. Leaving open for Nivedha or Avery.

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:195]: I wouldn't want to explain lex order either! If for some reason lex order comparisons are needed by a member, maybe chuck in a link/note directing readers to an in-depth explanation of it, since it has such a niche use-case?

File: frontend/javascript/README.md
Comment (justinquinnb) [reply frontend/javascript/README.md:328]: Will let Nivedha or Avery chime in on this.

**Comment by justinquinnb** [comment]
This has my utmost approval--looks great, very well done. Waiting on another set of eyes before we pull this in. Thank you for the great work!

File: frontend/javascript/README.md
Comment (ggliv) [reply frontend/javascript/README.md]: This is JavaScript, of course they do something unexpected. Jokes aside, I'll include a general statement on how operations can produce weird results and some very brief examples. I don't think it's important enough to go into much more detail than that.

File: frontend/javascript/README.md
Comment (ggliv) [reply frontend/javascript/README.md:173]: To be clear, this is very much a me thing. They behave similarly to other C langs, I just personally do not like the usage of increment and decrement. They aren't strictly necessary to do anything (just use `{+,-}=`), they only ever save like one line, and their behavior when used in ~anything outside of a for-loop is usually hard to grasp.

I'm still happy to include mention of them if others determine they need to be touched on, but IMO they don't.

File: frontend/javascript/README.md
Comment (ggliv) [reply frontend/javascript/README.md:205]: I think I'm comfortable letting people discover it themselves if they every need it. The JS resources listed at the bottom (or Google) should be enough for curious readers.

If anyone ever asks about it in the context of the club, we can add it pretty easily.

**Review by nivedhanatarajan7 — CHANGES_REQUESTED** [review]
Hi! The article looks great and is very thorough! I really appreciate the high level of detail you included in the article. I think the code examples you included are great. 

There are a lot of comments in these code examples, and while it is great, it may also make the article more difficult to read. I suggest separating the code examples into smaller blocks (ex. a block for the different examples within the if-else code example) and explain the blocks through comments and captions. 

Overall, the article is amazing! Thank you so much!

File: frontend/javascript/README.md
Comment (nivedhanatarajan7) [reply frontend/javascript/README.md:136]: Yes, adding the note about var as a footnote sounds fine. Thank you!

**Comment by ggliv** [comment]
Okay, I think I've addressed all of your review comments. I broke a few of the sections up where I could, but the conditionals section seemed mostly okay to me so I just extracted the bool ops/truthy section into its own block.

Thank you so much for taking the time to give this PR a look, I really appreciate it! Review work is far too underappreciated, I think.

**Review by nivedhanatarajan7 — APPROVED** [review]
The article looks great! Thank you so much!

**Review by justinquinnb — APPROVED** [review]

**Pull request merged**
